packages/headless/src/app/insight-engine/insight-engine.ts

```diff
@@ -80,6 +83,8 @@ export function buildInsightEngine(
     })
   );
 
+  engine.dispatch(fetchInterface());
```

Comment:
It might make sense to put this in a function (ie: `fetch()`), instead of when building the controller ?

That way you allow the user to determine when they want to retrieve and execute the network call.

For example, let's say for the preview feature, the implementer might want to fetch the config when the user change a configuration the interface, and not have to recreate the controller to fetch the new configuration.

Reply:
Good point, I'll move that into a separate controller.